src/js/oncoprint.js

```diff
@@ -0,0 +1,125 @@
+var Track = require('./track').Track;
+var _ = require('underscore');
+var d3 = require('d3');
+var $ = require('jquery');
+module.exports = {};
+
+// TODO: use self everywhere
+
+var defaultOncoprintConfig = {
+	cell_width: 10,
+	cell_padding: 3,
+	render: 'table',
+};
+
+function Oncoprint(container_selector_string, config) {
+	var self = this;
+	self.table;
+	self.config = $.extend({}, defaultOncoprintConfig, config || {});
+
+	self.id_order = [];
+	self.track_order = []; 
+	self.tracks = {};
+	self.ids = {};
+
+	if (self.config.render === 'table') {
```

Comment:
In some sense, can't `new OncoprintTableRenderer(container_selector_string, self)` itself be passed in as a `renderer` parameter to `Oncoprint`? This can allow us to put this implementation selection code elsewhere.